de.tudarmstadt.ukp.wikipedia.api/src/test/java/de/tudarmstadt/ukp/wikipedia/api/CategoryDescendantsIteratorTest.java

```diff
@@ -28,31 +28,24 @@
 import org.junit.BeforeClass;
 import org.junit.Test;
 
-import de.tudarmstadt.ukp.wikipedia.api.WikiConstants.Language;
 import de.tudarmstadt.ukp.wikipedia.api.exception.WikiApiException;
 
-public class CategoryDescendantsIteratorTest {
+public class CategoryDescendantsIteratorTest extends BaseJWPLTest{
 
-	private static Wikipedia wiki;
-
-	/**
+    /**
      * Made this static so that following tests don't run if assumption fails.
-     * (With AT_Before, tests would also not be executed but marked as passed)
-	 */
+     * (With AT_Before, tests also would not be executed but marked as passed)
+     * This could be changed back as soon as JUnit ignored tests after failed
+     * assumptions
+     */
 	@BeforeClass
 	public static void setupWikipedia() {
-		DatabaseConfiguration db = new DatabaseConfiguration();
-		db.setDatabase("wikiapi_test");
-		db.setHost("bender.ukp.informatik.tu-darmstadt.de");
-		db.setUser("student");
-		db.setPassword("student");
-		db.setLanguage(Language._test);
-		try {
-			wiki = new Wikipedia(db);
-		} catch (Exception e) {
-			assumeNoException(e);
-			//fail("Wikipedia could not be initialized.");
-		}
+        DatabaseConfiguration db = obtainHSDLDBConfiguration();
+        try {
+            wiki = new Wikipedia(db);
+        } catch (Exception e) {
```

Comment:
I wonder whether we should remove these try - catches as they might create the illusion that all tests are passing (specially for new contributors)

Reply:
I don't see why these are creating an illusion here. During test setup phase, exceptions might occur and that's why there is a try/catch here. Maybe, I don't get your concern here correctly.

Reply:
@mawiesne Unit tests should normally not contain try/catch statements. If an exception is thrown, JUnit should pick it up and mark the test as in error. The only reason to catch an exception would normally be to check its type/error message.

Reply:
@reckart Btw, this was the original code of the master branch, merely adapted. Shall I change it throughout all the other test cases in the api module?

Reply:
And a reason for #161 ...

Reply:
@mawiesne up to you, but if you do, I'd recommend doing it in a separate PR.

Reply:
@reckart you're right, code style is a topic to watch for. Once we have working (embedded) tests and this PR is merged, we can discuss/address changes related to this topic within #161. 
I see this PR as a first step in a row of pre-release actions. As already mentioned by @rzo1, testability should be improved, with it: code, code documentation, reduction of open TODOs/FIXMEs...

Reply:
@mawiesne wrt style cf https://github.com/dkpro/dkpro-jwpl/issues/164